src/dcqc/file.py

```diff
@@ -210,6 +196,25 @@ def _init_fs(self) -> tuple[FS, str]:
         self._fs = fs
         return fs, fs_path
 
+    @property
+    def local_path(self) -> Path:
```

Comment:
Nice refactor. Do you think we will need `@property.setters`?

Reply:
In the case of `local_path` and `name`, I don't want users to be able to change them. In the case of `local_path`, users can change it using `stage()`.